### PR TITLE
Employer Endpoints + Schema Validation

### LINKED ISSUE #48: Update & Delete Employer & Schema Validation
See [Database Schema Doc](https://github.com/uoftblueprint/tces/wiki/Database-Schema) for info.

API Endpoint info:

* DELETE `/api/employer/{employer_id}/`
* UPDATE `/api/employer/{employer_id}/`. Payload in the body should receive all the fields for a new employer object in the request body.

This should be split up into:

* a class in Model directory (Handle interacting with MongoDB directly)
* a class in Controller directory (This will use your class in Model directory to perform the expected functionality for each endpoint)
* a class in Routes directory (This will use functions from the class you define in the Controller directory, but actually tie them to the endpoint)

Also, add Schema Validation for `employer` in `conn.ts`. See [this tutorial ](https://www.mongodb.com/basics/json-schema-examples) for example.

See project structure
for further reference. Also msg me for clarification, I think this is a pretty confusing concept.

https://github.com/uoftblueprint/homes-for-heroes/tree/main/Backend/app

Reference this for examples, we have literally identical structure

This should build on your previous ticket

### PR DESCRIPTION
**Pull request by Manal-jpg**
Fixes #48 

## Sample POST
curl --location --request POST 'http://localhost:3001/api/employer' \
--header 'Content-Type: application/json' \
--data-raw '{
    "name": "John Doe",
    "primary_address_id": "63bb1f16939f28affbacf5fd",
    "contacts": [
        {
            "phone_num": "+1234567890",
            "email": "jdoe@acme.com"
        }
    ],
    "timeline": "63bb1f16939f28affbacf5fb",
    "creator_id": "63bb1f16939f28affbacf5fc",
    "owner_id": "63bb1f16939f28affbacf5fa",
    "legal_name": null,
    "hr_contacts": null,
    "secondary_address_id": null,
    "initial_contact": null
}'
## Response
Created an employer with id: 63fa9f6ba94e332cc3dd425b

## Sample DELETE Request
curl --location --request DELETE 'http://localhost:3001/api/employer/63dff4a5f2b1c7f2d0884bff'
## Reponse 
Deleted employer with id: 63dff4a5f2b1c7f2d0884bff

## Sample Update Request
curl --location --request PUT 'http://localhost:3001/api/employer/63faae7be501c2b1c5eb20d0' \
--header 'Content-Type: application/json' \
--data-raw '{
    "name": "John Papi",
    "primary_address_id": "63bb1f16939f28affbacf5fa",
    "contacts": [
        {
            "phone_num": "+1234567890",
            "email": "jdoe@acme.com"
        },
        {
            "phone_num": "+1234567890",
            "email": "jdoe@acme.com"
        },
        {
            "phone_num": "+1234567890",
            "email": "jdoe@acme.com"
        }
    ],
    "timeline": "63bb1f16939f28affbacf5fb",
    "creator_id": "63bb1f16939f28affbacf5fc",
    "owner_id": "63bb1f16939f28affbacf5fa"
}'

## Response 
Updated employer with id: 63faae7be501c2b1c5eb20d0

## Sample GET

curl --location --request GET 'http://localhost:3001/api/employer/63faae7be501c2b1c5eb20d0'

## Response

{
    "_id": "63faae7be501c2b1c5eb20d0",
    "name": "John Papi",
    "primary_address_id": "63bb1f16939f28affbacf5fa",
    "contacts": [],
    "creator_id": "63bb1f16939f28affbacf5fc",
    "owner_id": "63bb1f16939f28affbacf5fa",
    "legal_name": null,
    "hr_contacts": null,
    "secondary_address_id": null,
    "initial_contact": null,
    "timeline": {
        "notes": []
    },
    "NAICS": null
}
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [ ] I have run `npm run lint:fix`
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (if required)
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (Delete if frontend)

### CODE REVIEW

**Comment by ganeshasapu** [comment]
Can you pull from dev and resolve the conflicts, because I had to do some work on the employer files for my ticket

File: server/src/controllers/employer.controller.ts [hunk 1]
```diff
@@ -41,7 +40,44 @@ const createEmployer = async (req: Request, res: Response, next: NextFunction) =
...
-export { getEmployer, createEmployer };
+const deleteEmployer = async (req: Request, res: Response, next: NextFunction) => {
+
+    try { 
+        const id = req?.params?.id;
+        const result = await Employer.deleteEmployer(id);
+        res.status(202).send(result);
+    } catch (err) {
+        if (err instanceof Error) { 
+            console.log(err);
```
Comment (kenneth-miura) [inline server/src/controllers/employer.controller.ts:51]: please use the logger here, see https://github.com/uoftblueprint/tces/pull/85/files#diff-68408937f21e87d907b36faf633e1a024b1f9b2c3f1f6cf19572b7dfd5812b0f for example. Basically, it's that `error` function ganesh keeps calling

Comment (kenneth-miura) [reply server/src/controllers/employer.controller.ts:51]: basically anywhere that u are using console.log for an error, use the logger. It's more informative & better for debugging.

**Review by kenneth-miura — CHANGES_REQUESTED** [review]
looks great! Just one request regarding the use of the logger. Please msg me on slack once resolved so I don't end up blocking you.

**Pull request merged**